doc/modules/clustering.rst

```diff
@@ -1300,3 +1300,244 @@ Drawbacks
   concepts of clusters, such as density based clusters like those obtained
   through DBSCAN.
 
+
+Select number of clusters
+===============================
+
+.. figure:: ../auto_examples/cluster/images/plot_chosen_nb_cluster_comparaison.png
+   :target: ../auto_examples/cluster/plot_chosen_nb_cluster_comparaison.html
+   :align: center
+   :scale: 50
+
+   A comparison of algorithms to select the number of clusters in scikit-learn. The clustering algorithm used is spectral clustering
+
+.. currentmodule:: sklearn.metrics.cluster
+
+Many algorithms require to select the wanted number of clusters. If one does not know how many clusters he wants, there exists algorithm to find the most relevant number of clusters for its data, given the data and the clustering algorithm used.
```

Comment:
While number of clusters is the most common, (some of?) these techniques area generally interested in (hyper)parameter search for clustering algorithms.